tests/test_runlib.py

```diff
@@ -424,6 +424,23 @@ def test_hash_artifact_passing_algorithm(self):
     self.assertTrue("sha256" in list(_hash_artifact("foo", ["sha256"]).keys()))
 
 
+class TestLinkCmdExecTimeoutSetting(unittest.TestCase):
+  """Tests LINK_CMD_EXEC_TIMEOUT setting in settings.py file. """
+
+  def test_timeout_setting(self):
+    # Save the old timeout and make sure it's saved properly
+    timeout_old = in_toto.settings.LINK_CMD_EXEC_TIMEOUT
+    self.assertEqual(in_toto.settings.LINK_CMD_EXEC_TIMEOUT, timeout_old)
+
+    # Modify timeout and check that it returns the same value
+    timeout_new = timeout_old + 1
+    in_toto.settings.LINK_CMD_EXEC_TIMEOUT = timeout_new
+    self.assertEqual(in_toto.settings.LINK_CMD_EXEC_TIMEOUT, timeout_new)
+
+    # Restore original timeout
+    in_toto.settings.LINK_CMD_EXEC_TIMEOUT = timeout_old
```

Comment:
Hooray, for adding a test and making it work. 🎉  

Unfortunately, this currently only test if Python's name assignment works, which is not very meaningful for you newly added feature. Instead it would be great to test if the test new `LINK_CMD_EXEC_TIMEOUT` actually affects the timeout of an executed process. An easy way to do this, and this is how securesystemslib and Python's standard library subprocess module do it, is to use -1 as timeout parameter and see if `TimeoutExpired` is raised immediately.

Reply:
> is to use -1 as timeout parameter and see if TimeoutExpired is raised immediately.

@itsbenwu and @kristelfung arrived at that conclusion yesterday! :smile: